src/hotspot/share/memory/iterator.hpp

```diff
@@ -355,6 +355,10 @@ class SerializeClosure : public Closure {
   bool writing() {
     return !reading();
   }
+
+  // Useful aliases
```

Comment:
This is a nice addition but it might be buried under the rest of this PR. Would it be worth making this a separate enhancement?

Reply:
I filed [JDK-8307959](https://bugs.openjdk.org/browse/JDK-8307959): "Remove explicit type casts from SerializeClosure::do_xxx() calls"

 I will integrate that before integrate this PR.